datasets/datasets_decorator.py

```diff
@@ -27,6 +29,23 @@ def step_wrapper(*args, **kwargs):
                 _snake_name = _pascal_to_snake_case(dataset.name)
                 setattr(self, _snake_name, dataset)
 
+            # Transformation integration
+            if hasattr(self, "name"):
+                flow_name = self.name
+            else:
+                flow_name = type(self).__name__
+
+            add_txf_attributes(self)
+            if not self._txf_registered_datasets.get(flow_name, None):
+                self._txf_registered_datasets[flow_name] = [dataset]
```

Comment:
Why do we maintain a dictionary of `_txf_registered_datasets[flow_name]` where `flow_name` is always the name of the flow?

I think you mean `_txf_registered_datasets[dataset.name]`.   However, that may not work because one could reference the multiple Flow dataset variable/parameter to the same logically named `dataset.name`.

What is `_txf_registered_datasets` used for? Could you please reference a PR for what it is used for?